tests/rules/no-unused-collection.test.ts

```diff
@@ -197,11 +204,6 @@ ruleTester.run('Primitive return types should be used.', rule, {
           arrayConstructor[1] = 42;
       }
       `),
-    invalidTest(`function nok2_1() {
-          let arrayConstructor = new Array(); // Noncompliant
-          arrayConstructor[1][2] = 42;
-      }
```

Comment:
this would now be considered compliant code, as the arrayConstructor array is being read to access arrayConstructor[1], hence should be not considered unused.